Add support for psr/container ^2.0

Nguimjeu opened the pull request:
- Add support for `psr/container`: `^2.0`
- See: https://github.com/Roave/behat-psr11extension/issues/18

Fixes #18

Nguimjeu committed:
Add support for psr/container ^2.0

- Add support for `psr/container`: `^2.0`
- See: https://github.com/Roave/behat-psr11extension/issues/18

Ocramius commented on composer.json:
```diff
@@ -5,7 +5,7 @@
         "php": "^7.4 || ^8.0",
         "behat/behat": "^3.4",
-        "psr/container": "^1.0",
+        "psr/container": "^1.0 || ^2.0",
```
`composer.lock` needs updating too :thinking:

Nguimjeu replied:
@Ocramius Sorry for missing that. Latest commit addressing `composer.lock` update

aarseaingng committed:
Add support for psr/container ^2.0: update composer.lock

- Upgrade psr/container (1.1.2 => 2.0.2)
- Update function declaration for anonymous class in `ContainerFactoryTest.php`
- Run `composer update`

Ocramius reviewed (approved):
LGTM :+1: 

Thanks @Nguimjeu

Ocramius commented:
Ah, never mind, tests are all completely broken :D 

Nope, need to rework this until it's green, sorry.

Ocramius commented:
Needs https://github.com/laminas/laminas-servicemanager/pull/151

Ocramius commented:
Now handled in https://github.com/Roave/behat-psr11extension/releases/tag/2.6.0
